server/base/src/main/java/org/apache/accumulo/server/manager/state/ManagerTabletInfoIterator.java

```diff
@@ -0,0 +1,434 @@
+import org.apache.accumulo.core.iterators.IteratorEnvironment;
+import org.apache.accumulo.core.iterators.SkippingIterator;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.accumulo.core.iterators.user.WholeRowIterator;
+import org.apache.accumulo.core.manager.state.ManagerTabletInfo;
+import org.apache.accumulo.core.manager.state.ManagerTabletInfo.ManagementAction;
+import org.apache.accumulo.core.manager.thrift.ManagerState;
+import org.apache.accumulo.core.metadata.TServerInstance;
+import org.apache.accumulo.core.metadata.TabletState;
+import org.apache.accumulo.core.metadata.schema.DataFileValue;
+import org.apache.accumulo.core.metadata.schema.MetadataSchema.TabletsSection.ChoppedColumnFamily;
+import org.apache.accumulo.core.metadata.schema.MetadataSchema.TabletsSection.CurrentLocationColumnFamily;
+import org.apache.accumulo.core.metadata.schema.MetadataSchema.TabletsSection.FutureLocationColumnFamily;
+import org.apache.accumulo.core.metadata.schema.MetadataSchema.TabletsSection.HostingColumnFamily;
+import org.apache.accumulo.core.metadata.schema.MetadataSchema.TabletsSection.LastLocationColumnFamily;
+import org.apache.accumulo.core.metadata.schema.MetadataSchema.TabletsSection.LogColumnFamily;
+import org.apache.accumulo.core.metadata.schema.MetadataSchema.TabletsSection.SuspendLocationColumn;
+import org.apache.accumulo.core.metadata.schema.MetadataSchema.TabletsSection.TabletColumnFamily;
+import org.apache.accumulo.core.metadata.schema.TabletMetadata;
+import org.apache.accumulo.core.metadata.schema.TabletMetadata.Location;
+import org.apache.accumulo.core.metadata.schema.TabletMetadata.LocationType;
+import org.apache.accumulo.core.util.AddressUtil;
+import org.apache.hadoop.io.DataInputBuffer;
+import org.apache.hadoop.io.DataOutputBuffer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Sets;
+
+public class ManagerTabletInfoIterator extends SkippingIterator {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ManagerTabletInfoIterator.class);
+
+  private static final String SERVERS_OPTION = "servers";
+  private static final String TABLES_OPTION = "tables";
+  private static final String MERGES_OPTION = "merges";
+  private static final String DEBUG_OPTION = "debug";
+  private static final String MIGRATIONS_OPTION = "migrations";
+  private static final String MANAGER_STATE_OPTION = "managerState";
+  private static final String SHUTTING_DOWN_OPTION = "shuttingDown";
+
+  private static void setCurrentServers(final IteratorSetting cfg,
+      final Set<TServerInstance> goodServers) {
+    if (goodServers != null) {
+      List<String> servers = new ArrayList<>();
+      for (TServerInstance server : goodServers) {
+        servers.add(server.getHostPortSession());
+      }
+      cfg.addOption(SERVERS_OPTION, Joiner.on(",").join(servers));
+    }
```

Comment:
Think we can remove the null check.  This set comes from Manager.onlineTabletServers() which I linked in another comment and that never returns null.

```suggestion
      List<String> servers = new ArrayList<>();
      for (TServerInstance server : goodServers) {
        servers.add(server.getHostPortSession());
      }
      cfg.addOption(SERVERS_OPTION, Joiner.on(",").join(servers));

```

Reply:
I removed the null check in ManagerTabletInfoIterator.parseServers() and ended up with the following when running LocatorIT:

```
java.lang.NullPointerException: null
        at org.apache.accumulo.server.manager.state.ManagerTabletInfoIterator.parseServers(ManagerTabletInfoIterator.java:170) ~[classes/:?]
        at org.apache.accumulo.server.manager.state.ManagerTabletInfoIterator.init(ManagerTabletInfoIterator.java:292) ~[classes/:?]
        at org.apache.accumulo.core.iteratorsImpl.IteratorConfigUtil.loadIterators(IteratorConfigUtil.java:226) ~[classes/:?]

```

Reply:
Must be something with the serialization of an empty string or something

Reply:
I removed the null checks in TabletManagementIterator by making the collections final, and then adding the parsed information to them, in 9153726